src/ui/main.cpp

```diff
@@ -236,11 +237,11 @@ int main(int argc, char **argv) {
     DigitizerUi::BlockType::registry().loadBlockDefinitions(BLOCKS_DIR);
 #endif
 
-    DigitizerUi::DataSource::registerBlockType();
     DigitizerUi::DataSink::registerBlockType();
     DigitizerUi::DataSinkSource::registerBlockType();
-    DigitizerUi::ArithmeticBlock::registerBlockType();
-
+    DigitizerUi::BlockType::registry().addBlockType<opendigitizer::SineSource>("opendigitizer::SineSource");
+    DigitizerUi::BlockType::registry().addBlockType<opendigitizer::RemoteSource>("opendigitizer::RemoteSource");
+    DigitizerUi::BlockType::registry().addBlockType<opendigitizer::Arithmetic>("opendigitizer::Arithmetic");
```

Comment:
@ivan-cukic is there a way to run this function for every type known to BlockRegistry/PluginLoader? (with the actual type, not type-erased??). Or we would have to make sure all required metainformation is available via BlockModel.

Reply:
tbd later.